project2-beat-that/script.js

```diff
@@ -1,4 +1,337 @@
+
+//end of global variables
+
+var beatit2main = function (input) {
+  if (playerTurn_G2 === "Player1Rolls") {
+    player1Dice1_G2 = rollDice();
+    player1Dice2_G2 = rollDice();
+    playerTurn_G2 = "Player1Chooses";
+    console.log(`player 1 dice 1 is ${player1Dice1_G2}`);
+    console.log(`player 1 dice 2 is ${player1Dice2_G2}`);
+    console.log(playerTurn_G2);
+
+    textinputbox.style.display = "block";
+    togglebutton.innerHTML = "Submit!";
+
+    return `Player 1 rolls! <br>
+        First Dice 🎲: ${player1Dice1_G2} <br>
+        Second Dice 🎲: ${player1Dice2_G2}<br>
+        Please enter whether you want dice roll '1' or dice roll '2' to be the first number.`;
+  }
+  if (playerTurn_G2 === "Player1Chooses") {
+    //error validation
+    if (Number.isNaN(Number(input)) == true) {
+      return "sorry please enter a number.";
+    }
+    if (input < 1 || input > 2) {
+      return `Sorry, please enter either '1' or '2'<br>
+      Your First Dice 🎲: ${player1Dice1_G2} <br>
+      Your Second Dice 🎲: ${player1Dice2_G2}<br>
+      Please enter whether you want dice roll '1' or dice roll '2' to be the first number.`;
+    }
+    togglebutton.innerHTML = "Player 2 Roll Dice!";
+    textinputbox.style.display = "none";
+
+    // error validation ends
+    if (input == "1") {
+      player1FinalNumber_G2 = Number(`${player1Dice1_G2}${player1Dice2_G2}`);
+      console.log(player1FinalNumber_G2 + 1);
+      playerTurn_G2 = "Player2Rolls";
+      return `Player 1's final number is ${player1FinalNumber_G2}. <br> It is player 2's turn to roll your dice!`;
+    }
+    if (input == "2") {
+      player1FinalNumber_G2 = Number(`${player1Dice2_G2}${player1Dice1_G2}`);
+      console.log(player1FinalNumber_G2 + 1);
+      playerTurn_G2 = "Player2Rolls";
+      return `Player 1's final number is ${player1FinalNumber_G2}. <br> It is player 2's turn to roll your dice!`;
+    }
+  }
+
+  if (playerTurn_G2 === "Player2Rolls") {
+    player2Dice1_G2 = rollDice();
+    player2Dice2_G2 = rollDice();
+    playerTurn_G2 = "Player2Chooses";
+    console.log(`player 2 dice 1 = ${player2Dice1_G2}`);
+    console.log(`player 2 dice 1 = ${player2Dice2_G2}`);
+    textinputbox.style.display = "block";
+    togglebutton.innerHTML = "Submit!";
+
+    return `Player 2 rolls! <br>
+     First Dice 🎲: ${player2Dice1_G2} <br>
+     Second Dice 🎲: ${player2Dice2_G2}<br>
+     Please enter whether you want dice roll '1' or dice roll '2' to be your first number.`;
+  }
+
+  if (playerTurn_G2 === "Player2Chooses") {
+    //error validation
+    if (Number.isNaN(Number(input)) == true) {
+      return "sorry please enter a number.";
+    }
+    if (input < 1 || input > 2) {
+      return `Sorry, please enter either '1' or '2'<br>
+      Your First Dice 🎲: ${player1Dice1_G2} <br>
+      Your Second Dice 🎲: ${player1Dice2_G2}<br>
+      Please enter whether you want dice roll '1' or dice roll '2' to be the first number.`;
+    }
```

Comment:
maybe you can think of different game state conditions so you dont have to hard code each step of the game? so that player 1 and player 2 can use the same code?